docs/content/basic/app-assistant.md

```diff
@@ -0,0 +1,156 @@
+:::
+
+Agents and assistants comprise a new messaging experience for Slack. If you're unfamiliar with using agents and assistants within Slack, you'll want to read the [API documentation on the subject](https://api.slack.com/docs/apps/ai). Then, come back here to implement them with Bolt!
+
+## Configuring your app to support assistants 
+
+1. Within [App Settings](https://api.slack.com/apps), enable the **Agents & Assistants** feature.
+
+2. Within the App Settings **OAuth & Permissions** page, add the following scopes: 
+  * [`assistant:write`](https://api.slack.com/scopes/assistant:write)
+  * [`chat:write`](https://api.slack.com/scopes/chat:write)
+  * [`im:history`](https://api.slack.com/scopes/im:history)
+
+3. Within the App Settings **Event Subscriptions** page, subscribe to the following events: 
+  * [`assistant_thread_started`](https://api.slack.com/events/assistant_thread_started)
+  * [`assistant_thread_context_changed`](https://api.slack.com/events/assistant_thread_context_changed)
+  * [`message.im`](https://api.slack.com/events/message.im)
+
+:::info
+You _could_ implement your own assistants by [listening](/concepts/event-listening) for the `assistant_thread_started`, `assistant_thread_context_changed`, and `message.im` events. That being said, using the `Assistant` class will streamline the process. And we already wrote this nice guide for you!
+:::
+
+## The `Assistant` class instance
+
+```ts
+const assistant = new Assistant({
+  threadContextStore: {
+    get: async ({ context, client, payload }) => {},
+    save: async ({ context, client, payload }) => {},
+  },
+  threadStarted: async ({ say, saveThreadContext, setStatus, setSuggestedPrompts, setTitle }) => {},
+  threadContextChanged: async ({ say, setStatus, setSuggestedPrompts, setTitle }) => {},
+  userMessage: async ({ say, getThreadContext, setStatus, setSuggestedPrompts, setTitle }) => {},
+});
+```
+
+You can store context through the `threadContextStore` property but it must feature `get` and `save` methods. If not provided, a `DefaultThreadContextStore` instance is utilized instead, which is a reference implementation that relies on storing and retrieving message metadata as the context changes.
+
+## Handling a new thread
+
+The `threadStarted` event handler allows your app to respond to new threads opened by users. In the example below, the app is sending a message — containing context message metadata — to the user, along with a single [prompt](https://api.slack.com/methods/assistant.threads.setSuggestedPrompts).
+
+```js
+...
+  threadStarted: async ({ event, say, setSuggestedPrompts, saveThreadContext }) => {
+    const { context } = event.assistant_thread;
+
+    await say({
+      text: 'Hi, how can I help?',
+      metadata: { event_type: 'assistant_thread_context', event_payload: context },
```

Comment:
@misscoded Sorry, I might overlook this in code reviews but bolt-python/java automatically set this message metadata when a developers uses say within assistant listener code. Would it be possible to ask you to enhance createSay to embed metadata? https://github.com/slackapi/bolt-js/blob/198d7c317f901f051c636cf6786fa5633038c2c2/src/Assistant.ts#L302-L312

Reply:
> set this message metadata

No worries, I can see that change made. Can you advise or point me to what metadata you're referring to that's added? I thought the only metadata that we are adding automatically is to the parent Bot message in the thread. What does the additional metadata do?

Reply:
Indeed, appending only to the parent bot message is ideal, but it makes the implementation far more complex. So, what I did for python and java is to always attach the 'assistant_thread_context' one this way: https://github.com/slackapi/bolt-python/blob/v1.21.1/slack_bolt/context/assistant/assistant_utilities.py#L64-L74 For following bot messages, the metadata is never used but I don't think it can be a problem.

Reply:
note that the python's get_context_thread internally resolves the context using the payload for assistant_* events and fetches it from context store for messages: https://github.com/slackapi/bolt-python/blob/v1.21.1/slack_bolt/context/get_thread_context/get_thread_context.py

if bolt-js does not do the same, the equivalent code for bolt-js could be a bit different.

Reply:
Ah, ok, if I understand you correctly, this is only for the edge case when someone uses `say` and doesn't "hand-roll" the metadata attachment. Got it! I'll get that change in shortly.